### ISSUE TITLE
Please provider better/more restrictive AWS IAM policies 

### ISSUE THREAD

**Issue by loganhz** [rancher/docs#947]
Origin issue is https://github.com/rancher/rancher/issues/15744
**Rancher versions:**
rancher/server or rancher/rancher: 2.0.8

The example IAM policies for an AWS EC2 cluster are way too permissive. For anyone seriously running R2 in AWS, these policies shouldn't be used nor should they even be promoted outside of a development setting. The essentially give free reign on all ec2 resources.

It would be nice to get a subject matter expert from Rancher who understands what permissions are actually needed to help form IAM policies that are more specific.

An example of a more restrictive policy is the one I wrote for EBS volumes on R1 EBS Volume Plugin, it limits the capabilities of the node to generate volumes and then only delete the ones it created, as well as attach and detach nodes that are under rancher's control https://github.com/rancher/rancher/issues/9343#issuecomment-319468333

So far my attempts to use R2 K8S AWS Cloud Provider and provide a more restrictive IAM policy has resulted in many things not working. I finally got node creation to work, but I am unable to get EBS volumes to work.

**Comment by thapakazi** [comment]
**Anyone able to pull this off ?** 
I am having hard time creating a eks cluster with node, full run. This is 10+ time. &&  I am still wrestling via lots of iam issues. Also I can't afford to supply ADMIN access to provision the cluster.

**How do I retry the failed cluster provisioning ?**
 Also that I don't know how to manually invoke retry for failed cluster provisioning. Every-time I am deleting the cluster which invokes lots of things ( cleanups of cloudformation stacks {vpc,role,...}, eks cluster }. And  eventually it takes a lot of wait hours.

Such a great tool, and I searched the entire world and can't find the right resources for it. :disappointed:

**Comment by catherineluse** [comment]
The minimum EKS permissions are now documented here https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/hosted-kubernetes-clusters/eks/#appendix-minimum-eks-permissions

**Comment by catherineluse** [comment]
@SheilaghM More research needs to be done about what are the minimum IAM policies for deploying a cluster in EC2. I'm not sure how much it differs for what we have already documented about the minimum IAM policies for deploying clusters in EKS.

**Comment by divya-mohan0209** [comment]
Reviving this thread since I was combing through old issues. The [documentation](https://rancher.com/docs/rancher/v2.5/en/cluster-provisioning/hosted-kubernetes-clusters/eks/permissions/) clearly provides policies from an EKS standpoint. For EC2, @SheilaghM and @catherineluse , do you think it'd be beneficial to document it? If yes, who'd be a good POC to work with on the research?

**Comment by catherineluse** [comment]
Yes, I think it would be beneficial to document. The only reason we haven't already documented it is that is is difficult to find out what the minimum permissions are because you have to do a lot of trial-and-error testing with EC2 to see if you can provision EC2 clusters with restricted permissions. But the information is valuable to security-conscious users.

Eamon Baumon helped narrow down the list for EKS. Maybe he could also help narrow down the list for EC2?

We already have listed EC2 permissions that give users sufficient access: https://rancher.com/docs/rancher/v2.6/en/cluster-provisioning/rke-clusters/node-pools/ec2/#example-iam-policy The issue is about going through that list and removing anything that isn't strictly necessary.
